src/theory/quantifiers/sygus_sampler.cpp

```diff
@@ -627,12 +628,9 @@ Node SygusSampler::getRandomValue(TypeNode tn)
       Rational rr = r.getConst<Rational>();
       if (rr.sgn() == 0)
       {
-        return s;
-      }
-      else
-      {
-        return nm->mkConstReal(sr / rr);
+        return nm->mkConstReal(s.getConst<Rational>());
```

Comment:
Shouldn't we use `ensureReal()` or so here?

Reply:
`mkConstReal` will always return a `Real`, so this is not necessary.

Reply:
I meant to use `ensureReal()` instead of `mkConstReal()`.

Reply:
Ah I see. The reason I didn't do this is because ensureReal is a arith::rewriter utility, I'm not sure if we wanted to make these public.

Reply:
Ah I see.